src/app/shared/conditional-show/conditional-show.model.ts

```diff
@@ -30,4 +32,18 @@ export class ShowCondition {
   private removeStarChar(s: string) {
     return s.substring(0, s.length - 1);
   }
+
+  matchByCaseFields(caseFields: CaseField[]): boolean {
+    return this.match(this.buildState(caseFields));
+  }
+
+  private buildState(caseFields: CaseField[]): any {
```

Comment:
I think this whole method can be moved in FieldsUtils. Can we called toValuesMap or similar?
Also, perhaps we could reuse some method already there to achieve the same result